lib/ansible/inventory/group.py

```diff
@@ -246,8 +246,8 @@ def set_variable(self, key, value):
         if key == 'ansible_group_priority':
             self.set_priority(int(value))
         else:
-            if key in self.vars and isinstance(self.vars[key], MutableMapping) and isinstance(value, Mapping):
-                self.vars[key] = combine_vars(self.vars[key], value)
+            if key in self.vars:
```

Comment:
should we really remove the Mapping checks?

Reply:
I guess those checks still makes sense for efficiency - I thought at first it would change the behavior of merging hashes that contain lists, but combine_vars always replaces those.